vql/linux/audit/audit_service_test.go

```diff
@@ -63,28 +116,68 @@ func (self *TestListener) Receive(buf *auditBuf) error {
 }
 
 func (self *TestListener) Close() error {
+	if !self.opened || self.failClose {
+		return syscall.ENOTCONN
+	}
 	self.cancel()
+	self.cancel = nil
+	self.opened = false
 	return nil
 }
 
 func (self *AuditServiceTestSuite) SetupTest() {
 	self.ConfigObj = self.LoadConfig()
 
 	logger := logging.GetLogger(self.ConfigObj, &logging.ClientComponent)
-	listener := newTestListener()
-	client := newMockCommandClient()
-	self.auditService = newAuditService(self.ConfigObj, logger, listener, client)
+	self.listener = newTestListener()
+	self.client = newMockCommandClient()
+	self.auditService = newAuditService(self.ConfigObj, logger, self.listener, self.client)
 
 	self.TestSuite.SetupTest()
 }
 
 func (self *AuditServiceTestSuite) TearDown() {
```

Comment:
This is not running. Looking at the other tests in the repo, I think it should this be `TearDownTest() ` and call `self.TestSuite.TearDownTest() first`. Then `TestRunService()` fails.

Reply:
yep